tensorflow/python/compiler/tensorrt/test/quantization_mnist_test.py

```diff
@@ -281,5 +299,110 @@ def testEval(self):
     self.assertAllClose(0.9675, accuracy_tf_trt, rtol=1e-3, atol=1e-3)
 
 
+class MNISTTestV2(QuantizationAwareTrainingMNISTTest):
+
+  def _SaveModel(self, model_dir, output_dir):
+    saved_model_builder = builder.SavedModelBuilder(output_dir)
+    graph = ops.Graph()
+    with Session(graph=graph) as sess:
+      with graph.device('/GPU:0'):
+        x = array_ops.placeholder(
+            shape=(None, 28, 28, 1), dtype=dtypes.float32, name=INPUT_NODE_NAME)
+        self._BuildGraph(x)
+      self._LoadWeights(model_dir, sess)
+      input_tensor = graph.get_tensor_by_name(INPUT_NODE_NAME + ":0")
+      output = graph.get_tensor_by_name(OUTPUT_NODE_NAME + ":0")
+      signature_def = signature_def_utils.build_signature_def(
+            inputs={"input": saved_model_utils.build_tensor_info(input_tensor)},
+            outputs={"output": saved_model_utils.build_tensor_info(output)},
+            method_name=signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY)
+      saved_model_builder.add_meta_graph_and_variables(
+            sess, [tag_constants.SERVING],
+            signature_def_map={
+                signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY:
+                signature_def
+            })
+    saved_model_builder.save()
+
+  def _GetFunc(self, use_trt, model_dir, use_dynamic_shape):
+    """Get the mnist function.
+
+    Args:
+      use_trt: whether use TF-TRT to convert the graph.
+      model_dir: the model directory to load the checkpoints.
+      use_dynamic_shape: whether to run the TF-TRT conversion in dynamic shape
+        mode.
+
+    Returns:
+      The frozen mnist GraphDef.
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+      saved_model_dir = os.path.join(tmpdir, 'mnist')
+      self._SaveModel(model_dir, saved_model_dir)
+
+      if use_trt:
+        conv_params=trt_convert.TrtConversionParams(
+            precision_mode='FP16', minimum_segment_size=2,
+            max_workspace_size_bytes=1<<28, maximum_cached_engines=1)
+        converter = trt_convert.TrtGraphConverterV2(
+            input_saved_model_dir=saved_model_dir,
+            conversion_params=conv_params,
+            use_dynamic_shape=use_dynamic_shape,
+            dynamic_shape_profile_strategy='ImplicitBatchModeCompatible')
+        converter.convert()
+        func = converter._converted_func
+      else:
+        saved_model_loaded = saved_model_load(
+            saved_model_dir, tags=[tag_constants.SERVING])
+        func = saved_model_loaded.signatures[
+            signature_constants.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
+    return func
+
+  def _Run(self, use_trt, batch_size, model_dir, use_dynamic_shape=False):
+    """Train or evaluate the model.
```

Comment:
Does this routine only "evaluate" but not "train" the model?
Looks like we don't follow the rule "using third person singular verb forms in subroutine document" in this whole file.